catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -894,20 +886,28 @@ private Object interpretObjectSensor(Sensors sensor, Sprite sprite) {
 		return returnValue;
 	}
 
-	private Double interpretOperatorEqual(Object left, Object right) {
+	private Double interpretOperatorCompare(Object left, Object right) {
 		try {
 			Double tempLeft = Double.valueOf(String.valueOf(left));
 			Double tempRight = Double.valueOf(String.valueOf(right));
 			int compareResult = getCompareResult(tempLeft, tempRight);
-			if (compareResult == 0) {
+			if (compareResult < 0) {
 				return 1d;
 			}
+			else if (compareResult > 0) {
+				return 2d;
+			}
 			return 0d;
+
 		} catch (NumberFormatException numberFormatException) {
-			int compareResult = String.valueOf(left).compareTo(String.valueOf(right));
-			if (compareResult == 0) {
+			int compareResult = String.valueOf(left).toLowerCase().compareTo(String.valueOf(right).toLowerCase());
+
+			if (compareResult < 0) {
 				return 1d;
 			}
+			else if (compareResult >0){
+				return 2d;
+			}
```

Comment:
Please change the return type to `int` and return
* 0 if left is equal to right;
* a value less than 0 if left is less than right;
* and a value greater than 0 if left is greater than right.

This is the default definition of compare methods.